src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OpenIDCallback.java

```diff
@@ -0,0 +1,37 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc;
+
+import java.io.IOException;
+
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+@WebServlet("/oidc/callback/*")
```

Comment:
I wrote a longer comment at the `OpenIdConfigBean.getRedirectUri()` method. I question the necessity of this URL router.

Reply:
I think it depends if we want to allow cookies or not. If we assume that nobody wants to use httponly secure cookies as being OK for security and always implement local storage/in memory tokens, like many systems do, then we can delete this routing thing. I just like the simplicity of it. You cannot turn the cookies off anyway if you want to use the OIDC annotation on payara, so we can have some convenience code as well. I see it more like that: either we happily use what is possible on Payara/Jakarta, or we implement everything ourselves, like it was the case until now. If we do use the OIDC annotation, we might as well use the convenience of it and not hide it under a rug for some hacker to figure it out anyway.